### PR TITLE
Update to v2

### PR DESCRIPTION
**Pull request by eduardoboucas**
Right, so here's a bunch of updates. Amongst other things:
- Support for multiple sites (called _properties_) — #32 
- Introduction of a `_id` field that gets appended to every entry
- Simple generated fields: when an element of `generatedFields` isn't an object, its value is used directly. This can be used to append fields to data fields, like strings, numbers or arrays.
- Schema files (using [Convict](https://github.com/mozilla/node-convict)) for both the API config (`config.js`) and the site config (`siteConfig.js`)
- Introduction of encrypted fields to the config: this can be achieved by setting a config field to use the type `'EncryptedString'` in the schema file. When the property is accessed using `this.siteConfig.get()`, the value will be decrypted automatically.
- `slugify` generated field

@zburgermeiszter here it is, as promised!

### CODE REVIEW

**Comment by eduardoboucas** [comment]
I want to do some more testing before I merge this to the main branch. In the meantime, you can test this (and whatever is deployed to the `dev` branch) on the dev instance, which is located at https://dev.staticman.net.

**Comment by zburgermeiszter** [comment]
Woohoo. Sounds good :)
I'll have a look at it.

File: lib/Staticman.js [hunk 1]
```diff
@@ -154,24 +241,60 @@ Staticman.prototype._getExtensionForFormat = function (format) {
...
+  }
+}
+
+Staticman.prototype._processSubscriptions = function (fields, options) {
+  // Initialise SubscriptionsManager
+  const subscriptions = new SubscriptionsManager(this.github, this.rsa)
+
+  // If there isn't a parent or if notifications are not enabled, there's
+  // nothing to do here
+  if (!options.parent || !this.siteConfig.get('notifications.enabled')) {
```
Comment (zburgermeiszter) [inline lib/Staticman.js:265]: What about people who want to subscribe other unrelated threads?

If it's real need, people will open an issue, so don't worry about it, just keep in mind.

Comment (eduardoboucas) [reply lib/Staticman.js:265]: The code for notifications is still WIP.

In any case, the idea there is that if you want to subscribe to something, you need to send the ID of the comment you're subscribing to. That will come through as `options[parent]`.

Comment (zburgermeiszter) [reply lib/Staticman.js:265]: Thanks.

**Pull request merged**